go-controller/pkg/ovn/ovn.go

```diff
@@ -1051,6 +1051,23 @@ func (oc *Controller) GetServiceVIPToName(vip string, protocol kapi.Protocol) (t
 	return namespace, ok
 }
 
+// GetPolicyLogging retrieves true if ACL policy logging is enabled for this Namespace
+func (oc *Controller) GetPolicyLogging(ns string) bool {
+	nsInfo := oc.getNamespaceLocked(ns)
+	if nsInfo == nil {
+		return false
+	}
+	defer nsInfo.Unlock()
+
+	for _, np := range nsInfo.networkPolicies {
+		if np.aclLogging {
+			return true
```

Comment:
It seems weird to specify logging on NetworkPolicies but then have the annotation on a single NetworkPolicy then be applied to the namespace as a whole. If it's going to apply to the namespace, it should be a namespace annotation.

Reply:
More importantly; as currently written, this feature confuses the different levels of privilege within the cluster; NetworkPolicies are controlled at the project level, by individual developers deploying applications onto the cluster, so it doesn't make sense for them to be requesting logs that would only be visible to cluster administrators, which the cluster administrators don't even want.

If the functionality is going to be controlled at the NetworkPolicy level, then there needs to be a global switch to enable/disable the feature, because we'd want to disable it in OpenShift.

(We're interested in policy-related logging in OpenShift, but I don't think this feature is the right implementation. At the moment, my theory is that something making use of OVS's NetFlow/IPFIX support would be more useful.)

Reply:
yes, I do not like mixing it either. let me think wrt. better approach. I think at the end we need something built-in anyways... ala calico

Reply:
@danwinship @girishmg as per discussion, made this feature optional and moved configuration to be per-namespace. Scope is also reduced to support deny/reject logging only with a path for us to add allow* in the future. PTAL w/ you have time.

Reply:
@girishmg rebased, retested and added topology upgrade. PTAL when you have time. Also, @astoycos it may affect #1792